husc/screens/cellomics.py

```diff
@@ -0,0 +1,366 @@
+
+    Returns
+    -------
+    stitched_image : array, shape (5*M, 5*N)
+        The stitched image.
+    """
+    fns.sort()
+    order = [[20, 21, 22, 23, 24],
+             [19, 6, 7, 8, 9],
+             [18, 5, 0, 1, 10],
+             [17, 4, 3, 2, 11],
+             [16, 15, 14, 13, 12]]
+
+    image0 = io.imread(fns[0])
+    m = image0.shape[0]
+    n = image0.shape[1]
+    stitched_image = np.zeros((5*m, 5*n))
+    for i in range(0, 5):
+        for j in range(0, 5):
+            index = order[i][j]
+            image = io.imread(fns[index])
+            stitched_image[m*i:m*(i+1), n*j:n*(j+1)] = image
+    return stitched_image
+
+
+def make_key2file(fns):
+    """Return a dictionary mapping well co-ordinates to filenames.
+
+    Returns a dictionary where key are (plate, well) co-ordinates and
+    values are lists of images corresponding to that plate and well.
+
+    Parameters
+    ----------
+    fns : list of string
+        A list of Cellomics TIF files.
+
+    Returns
+    -------
+    wellchannel2file : dict {tuple : list of string}
+        The dictionary mapping the (plate, well) co-ordinate to
+        a list of files corresponding to that well.
+    """
+    wellchannel2file = coll.defaultdict(list)
+    for fn in fns:
+        fn_base = os.path.basename(fn)
+        file_info = cellomics_semantic_filename(fn_base)
+        coord = (file_info['plate'], file_info['well'])
+        wellchannel2file[coord].append(fn)
+    return wellchannel2file
+
+
+def get_by_ext(path, extension, full=True):
+    """Return list of files in directory with specified extension.
+
+    Parameters
+    ----------
+    dir : string
+        A directory containing files.
+    extension : string
+        Return only files with this extension.
+    full : bool
+        Whether or not to return files with the path included. Default
+        true.
+    sort : bool
+        Whether or not to sort the list of files before returning them.
+        Default true.
+    """
+    fns = []
+    for dirpath, dirnames, filenames in os.walk(path):
+        for filename in [f for f in filenames if f.endswith("." + extension)]:
+            fns.append(os.path.join(dirpath, filename))
+    return fns
+
+
+def channel(fn):
+    """Get channel from Cellomics filename.
+
+    Parameters
+    ----------
+    fn : string
+        A filename from the Cellomics high-content screening system.
+
+    Returns
+    -------
+    channel : int
+        The channel of the filename.
+
+    Examples
+    --------
+    >>> fn = 'MFGTMP_140206180002_A01f00d0.TIF'
+    >>> channel(fn)
+    0
+    """
+    sem = cellomics_semantic_filename(fn)
+    channel = sem['channel']
+    return channel
+
+
+def column(fn):
+    """Get column from Cellomics filename.
+
+    Parameters
+    ----------
+    fn : string
+        A filename from the Cellomics high-content screening system.
+
+    Returns
+    -------
+    column : int
+        The channel of the filename.
```

Comment:
I added this because I was excluding columns 01 and 24 when doing the feature extraction on 10A-LNC-SP and thought it might be useful. Probably not because we since discussed adding filtering features to the interface :)